HW2/P3/mandelbrot.pyx

```diff
@@ -42,7 +42,88 @@ cpdef mandelbrot(np.complex64_t [:, :] in_coords,
                     z = z * z + c
                 out_counts[i, j] = iter
 
+# use instruction level
+cdef void counts_to_output(AVX.float8 counts,
+                           np.uint32_t [:, :] out_counts,
+                           int i,
+                           int j) nogil:
+    cdef:
+        float tmp_counts[8]
+        int k
+
+    AVX.to_mem(counts, &(tmp_counts[0]))
+
+    for k in range(8):
+        out_counts[i, j*8+k]=<unsigned int>tmp_counts[k]
+
+cpdef mandelbrot_new(np.complex64_t [:, :] in_coords,
+                     np.uint32_t [:, :] out_counts,
+                     int max_iterations=511):
+    cdef:
+       int i, j, iter, k
+       float [:, :] real, imag
+       AVX.float8 cr, ci, zr, zr_n, zi, magz2, signs, counts
```

Comment:
give variables either a short explanation or a meaningful name. I.e. I guessed zr = real part of z and an 8 vector but I would be lost for cr